groupActions/groupNotifier.go

```diff
@@ -146,6 +108,42 @@ func (gn *groupNotifier) notifyGroupsForTrigger(triggerID string, header data.He
 	}
 }
 
+// IsInterfaceNil returns true if the receiver is nil, false otherwise
+func (gn *groupNotifier) IsInterfaceNil() bool {
+	return gn == nil
+}
+
+// registerTrigger registers a new trigger that can be used by groups to be notified on it's triggered events
+// the parameter validation needs to be done by caller
+func (gn *groupNotifier) registerTrigger(trigger groupTypes.RegistrationHandler) error {
+	notifyOrder := gn.computeNextNotifyOrder()
+	gnt, err := newGroupNotifierTrigger(trigger.GetName(), notifyOrder, gn)
+	if err != nil {
+		return err
+	}
+
+	return gn.addTriggerWithGroupNotifier(trigger, gnt)
+}
+
+// parameter validation needs to be done by caller
+func (gn *groupNotifier) addTriggerWithGroupNotifier(trigger groupTypes.RegistrationHandler, notifierTrigger *groupNotifierTrigger) error {
```

Comment:
This function could never return an error.
Perhaps refactor and below instead of 

```go
	if exists {
		return nil
	}
```

You can:

```go
	if exists {
		return
	}
```